src/nvim/hardcopy.c

```diff
@@ -728,13 +730,10 @@ void ex_hardcopy(exarg_T *eap)
           if (got_int || settings.user_abort)
             goto print_fail;
 
-          sprintf((char *)IObuff, _("Printing page %d (%d%%)"),
-              page_count + 1 + side,
-              prtpos.bytes_printed > 1000000
-              ? (int)(prtpos.bytes_printed /
-                      (bytes_to_print / 100))
-              : (int)((prtpos.bytes_printed * 100)
-                      / bytes_to_print));
+          assert(prtpos.bytes_printed * 100 > prtpos.bytes_printed);
```

Comment:
Original code tried to do calculation in two different ways in order to expand the range of values for which calculation could be done without overflow.
Now that prtpos.bytes_printed is size_t, I think range is big enough. So, I simplified calculation to be done just one way, after checking for improbable overflow. I think that's clearer.